pkg/util/kms/kms_azure.go

```diff
@@ -0,0 +1,121 @@
+package kms
+
+import (
+	"fmt"
+
+	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+
+	"github.com/libopenstorage/secrets"
+	"github.com/libopenstorage/secrets/azure"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Azure authentication config options
+const (
+	AzureVaultURL       = "AZURE_VAULT_URL"
+	AzureVaultClientID  = "AZURE_CLIENT_ID"
+	AzureVaultTenantID  = "AZURE_TENANT_ID"
+	ServiceName         = "KMS_SERVICE_NAME"
+	AzureClientCertPath = "AZURE_CERT_SECRET_NAME"
+)
+
+// AzureVault is a azure kms driver
+type AzureVault struct {
+	UID string // NooBaa system UID
+}
+
+// NewAzureVault is azure driver constructor
+func NewAzureVault(
+	name string,
+	namespace string,
+	uid string,
+) Driver {
+	return &AzureVault{uid}
+}
+
+//
+// Azure vault specific driver for root master key
+//
+
+// Config returns this driver secret config
+func (*AzureVault) Config(config map[string]string, tokenSecretName, namespace string) (map[string]interface{}, error) {
+	// create a type correct copy of the configuration
+	c := make(map[string]interface{})
+	for k, v := range config {
+		c[k] = v
+	}
+
+	// create TLS files out of secrets
+	// replace in azure vaultConfig secret names with local temp files paths
```

Comment:
from where do we know that we expect a file path and not just the string itself?

Reply:
That is what I found in the secrets: https://github.com/libopenstorage/secrets/blob/5f4b25ceede180e8d67577288bba4223af658eb1/azure/azure_kv.go#L64
Am I missing something?